common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt

```diff
@@ -75,15 +81,21 @@ class HTTPClient(
     fun performRequest(
         path: String,
         body: Map<String, Any?>?,
-        headers: Map<String, String>
-    ): Result {
+        authenticationHeaders: Map<String, String>
+    ): HTTPResult {
         val jsonBody = body?.convert()
 
         val fullURL: URL
         val connection: HttpURLConnection
+        val httpRequest: HTTPRequest
         try {
             fullURL = URL(appConfig.baseURL, "/v1$path")
-            connection = getConnection(fullURL, headers, jsonBody)
+            val headersWithoutETag = getHeaders(authenticationHeaders)
+            httpRequest = HTTPRequest(fullURL, headersWithoutETag, jsonBody).let {
+                eTagManager.addETagHeaderToRequest(it)
```

Comment:
addETagHeader could make a nice extension function on `HTTPRequest`

Reply:
I initially did it an extension function, but then I realize I still need the instance of `eTagManager` (it has the shared preferences and the hashes map). The extension function would be something like `request.addETagHeader(eTagManager)`, which I don't think looks better than this. That, plus the fact that mocking extension functions is cumbersome, I would prefer to have the way it is now, `eTagManager.addETagHeaderToRequest`

Reply:
very fair, have not had to think about mocking extension functions before 😅